packages/filebrowser/src/model.ts

```diff
@@ -104,7 +104,23 @@ export class FileBrowserModel implements IDisposable {
     };
     window.addEventListener('beforeunload', this._unloadEventListener);
     this._scheduleUpdate();
-    this._startTimer();
+    this._poll = new Poll({
+      factory: async () => {
+        if (this._requested) {
+          return this.refresh();
+        }
+        let date = new Date().getTime();
+        if (date - this._lastRefresh > refreshInterval) {
+          return this.refresh();
+        }
```

Comment:
All of this logic can go away with the new Poll class, I think. It's sufficient to just call the method. When we want to have the poll immediately refresh, we call the poll's refresh method.

Reply:
Ah, good point. It also does some logic around a minimum refresh interval, which I tried to keep intact.

Reply:
We used to have a minimum interval option in Poll, but it didn't handle manual refreshes (it assumed a manual refresh was always immediate). Perhaps we add a minimum refresh interval to Poll for this debouncing. I think it would be pretty straightforward - the poll refresh() method would just schedule using logic like here.

So this logic isn't really a true debounce, in that it doesn't schedule a refresh according to MIN_REFRESH, it simply ignores requests if they are too soon after the last tick. That's perhaps a little frustrating - you click and click and click, and if you stop clicking before MIN_REFRESH, it's like you never clicked at all.

Reply:
actually, it's a bit more complicated for Poll, since we *really* want to debounce the callback (and every time we are about to call the callback, we check to see if we should go into standby, etc.). So that means the Poll should really be storing the last time it called the callback and make sure it's not calling the callback more often than it should.